app/src/main/java/com/example/returnpals/viewmodel/LoginViewModel.kt

```diff
@@ -20,16 +23,16 @@ import kotlin.coroutines.CoroutineContext
 //Login View model provides the information and function needed to login, logout, and signup.
 class LoginViewModel(
     email: String = "",
-    password: String = ""
+    password: String = "",
+    navController: NavController
 //    private val repository: LoginRepository       login repo is a global object for now
 ): ViewModel() {
-
     /** Gets updated on calls to [logIn], [logOut], [register], and [logInAsGuest]. **/
     val isLoggedIn get() = LoginRepository.isLoggedIn
     val isGuest get() = LoginRepository.isGuest
     var email by mutableStateOf(email)
     var password by mutableStateOf(password)
-
+    val navController = navController
```

Comment:
I agree. I think we first need to get the navController out of every nested composable function. I thought the view model might be a good place to keep it but it could also stay in the higher-level composable function. The issue is also how the logic for navigating is tightly coupled with the UI and that why the navController is passed down into nested function calls.  If I can get all the navController into the view models I am sure it will be easy to move them all out in the same way. At least then you would know their location.

Reply:
IMO, if everything was ideal, _all_ the navigation would stay in `NavHost.kt`, so that there would only be one central location for all the navigation, and no navcontroller would be allowed to leave the NavHost composable.

This is a 180 from our last convo, I thought you were settled on navcontrollers in viewmodels, its OK to disagree with me